src/rpc/auth.rs

```diff
@@ -8,4 +8,249 @@ pub use authpb::Permission;
 pub use authpb::Role;
 pub use authpb::User;
 pub use authpb::UserAddOptions;
-pub use etcdserverpb::auth_client::AuthClient;
+pub use etcdserverpb::auth_client::AuthClient as PbAuthClient;
+
+use etcdserverpb::{
+    AuthDisableRequest as PbAuthDisableRequest, AuthDisableResponse as PbAuthDisableResponse,
+    AuthEnableRequest as PbAuthEnableRequest, AuthEnableResponse as PbAuthEnableResponse,
+    AuthenticateRequest as PbAuthenticateRequest, AuthenticateResponse as PbAuthenticateResponse,
+};
+
+use crate::error::Result;
+use crate::rpc::ResponseHeader;
+use std::string::String;
+use tonic::transport::Channel;
+use tonic::{Interceptor, IntoRequest, Request};
+
+/// Client for Auth operations.
+#[repr(transparent)]
+pub struct AuthClient {
+    inner: PbAuthClient<Channel>,
+    // token: String
+}
+
+impl AuthClient {
+    /// Creates an auth client.
+    #[inline]
+    pub fn new(channel: Channel, interceptor: Option<Interceptor>) -> Self {
+        let inner = match interceptor {
+            Some(it) => PbAuthClient::with_interceptor(channel, it),
+            None => PbAuthClient::new(channel),
+        };
+
+        Self { inner }
+    }
+
+    /// auth_enable enables authentication.
+    #[inline]
+    pub async fn auth_enable(&mut self) -> Result<AuthEnableResponse> {
+        let options: Option<AuthEnableOptions> = Some(AuthEnableOptions::new());
+        let resp = self
+            .inner
+            .auth_enable(options.unwrap_or_default())
+            .await?
+            .into_inner();
+        Ok(AuthEnableResponse::new(resp))
+    }
+
+    /// auth_disable disables authentication.
+    #[inline]
+    pub async fn auth_disable(&mut self) -> Result<AuthDisableResponse> {
+        let options: Option<AuthDisableOptions> = Some(AuthDisableOptions::new());
+        let resp = self
+            .inner
+            .auth_disable(options.unwrap_or_default())
+            .await?
+            .into_inner();
+        Ok(AuthDisableResponse::new(resp))
+    }
+
+    /// Authenticate processes an authenticate request.
+    #[inline]
+    pub async fn authenticate(
+        &mut self,
+        name: String,
+        password: String,
+    ) -> Result<AuthenticateResponse> {
+        let options: Option<AuthenticateOptions> = Some(AuthenticateOptions::new());
+        let resp = self
+            .inner
+            .authenticate(options.unwrap_or_default().with_user(name, password))
+            .await?
+            .into_inner();
+        Ok(AuthenticateResponse::new(resp))
+    }
+}
+
+/// Options for `AuthEnable` operation.
+#[derive(Debug, Default, Clone)]
+// #[repr(transparent)]
```

Comment:
Why comments `repr`?

Reply:
It will due to compile error.
```
transparent struct needs exactly one non-zero-sized field, but has 0

needs exactly one non-zero-sized field, but has 0rustc(E0690)
auth.rs(79, 1): needs exactly one non-zero-sized field, but has 0
```